portable/MSVC-MingW/portmacro.h

```diff
@@ -143,7 +147,13 @@ void vPortExitCritical( void );
 
 /* BitScanReverse returns the bit position of the most significant '1'
  * in the word. */
+#if defined( __x86_64__ ) || defined( _M_X64 )
+        #define portGET_HIGHEST_PRIORITY( uxTopPriority, uxReadyPriorities )    uxTopPriority = 0; _BitScanReverse64( ( DWORD * ) &( uxTopPriority ), ( uxReadyPriorities ) )
```

Comment:
Thanks for your earlier comment about setting `uxTopPriority` to 0.  The pointer casting performed here is the root cause of the issue and should be resolved directly -- ie, don't cast a uint64_t pointer as a uint32_t pointer for the call to `_BitScanReverse64()`.  Both `_BitScanReverse()` and `_BitScanReverse64()` write the result to a 32-bit integer. Something like this would resolve it:

```
#define portGET_HIGHEST_PRIORITY( uxTopPriority, uxReadyPriorities ) \
    do                                                               \
    {                                                                \
        DWORD ulTopPriority;                                         \
        _BitScanReverse64( &ulTopPriority, ( uxReadyPriorities ) );  \
        uxTopPriority = ulTopPriority;                               \
    } while( 0 )
```

Also best if you can indent the new conditionals following the pattern established above these changes.

Reply:
Done. Also, thank you for the explanation.